src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1459 @@
+"""Module for SOP Classes of Presentation State (PR) IODs."""
+from collections import defaultdict
+import datetime
+from typing import Sequence, Optional, Tuple, Union
+
+from PIL.ImageCms import ImageCmsProfile, createProfile
+
+from pydicom import Dataset
+from pydicom.sr.coding import Code
+from pydicom.uid import ExplicitVRLittleEndian
+from pydicom._storage_sopclass_uids import (
+    GrayscaleSoftcopyPresentationStateStorage,
+    PseudoColorSoftcopyPresentationStateStorage,
+    ColorSoftcopyPresentationStateStorage
+)
+from pydicom.valuerep import DA, PersonName, TM, format_number_as_ds
+
+import numpy as np
+
+from highdicom.base import SOPClass
+from highdicom.content import (
+    ContentCreatorIdentificationCodeSequence,
+    LUT,
+    ModalityLUT,
+    PaletteColorLookupTable,
+    ReferencedImageSequence,
+)
+from highdicom.pr.content import (
+    GraphicLayer,
+    GraphicGroup,
+    GraphicAnnotation,
+    SoftcopyVOILUT,
+)
+from highdicom.pr.enum import PresentationLUTShapeValues
+from highdicom.enum import RescaleTypeValues
+from highdicom.sr.coding import CodedConcept
+from highdicom.uid import UID
+from highdicom.valuerep import (
+    check_person_name,
+    _check_code_string,
+    _check_long_string
+)
+
+
+class _SoftcopyPresentationState(SOPClass):
+
+    """An abstract base class for various Presentation State objects."""
+
+    def __init__(
+        self,
+        referenced_images: Sequence[Dataset],
+        series_instance_uid: str,
+        series_number: int,
+        sop_instance_uid: str,
+        sop_class_uid: str,
+        instance_number: int,
+        manufacturer: str,
+        manufacturer_model_name: str,
+        software_versions: Union[str, Tuple[str]],
+        device_serial_number: str,
+        content_label: str,
+        content_description: Optional[str] = None,
+        graphic_annotations: Optional[Sequence[GraphicAnnotation]] = None,
+        graphic_layers: Optional[Sequence[GraphicLayer]] = None,
+        graphic_groups: Optional[Sequence[GraphicGroup]] = None,
+        concept_name_code: Union[Code, CodedConcept, None] = None,
+        institution_name: Optional[str] = None,
+        institutional_department_name: Optional[str] = None,
+        content_creator_name: Optional[Union[str, PersonName]] = None,
+        content_creator_identification: Optional[
+            ContentCreatorIdentificationCodeSequence
+        ] = None,
+        rescale_intercept: Union[int, float, None] = None,
+        rescale_slope: Union[int, float, None] = None,
+        rescale_type: Union[RescaleTypeValues, str, None] = None,
+        modality_lut: Optional[ModalityLUT] = None,
+        copy_modality_lut: bool = False,
+        softcopy_voi_luts: Optional[Sequence[SoftcopyVOILUT]] = None,
+        icc_profile: Optional[ImageCmsProfile] = None,
```

Comment:
Let's not include the Pillow type in the public API.

```suggestion
        icc_profile: Optional[bytes] = None,
```

Reply:
Addressed in e6ba4fa